src/screens/session.tsx

```diff
@@ -39,7 +41,18 @@ export default function Session() {
       </Box>
       <Box>
         <Container fit header="Sessions">
-          TODO
+          <Stack gap="16px">
+            {Object.values(sessions).map((session) => {
```

Comment:
I think sessions is already an array so don't need the extra `Object.values` since your getSessions method does it!

Reply:
Actually if you switch this back to the older code 

```ts
const { sessions } = useSessionsStore()
const isConnected = Boolean(host && getSession({ host }))
``` 

and keep the Object.values here, it will update as expected. I believe your `getSessions()` isn't tracked like `sessions` since it doesn't re-render the `isConnected` button either until you click out

Reply:
Ah, I forgot the `Object.values` there because I was trying exactly that. For some reason that doesn't work either when I run it locally. It does re-render when you connect/disconnect to the host for the current tab, but not if you remove another session. You can reproduce it by applying this diff below and then try to remove a session that isn't the current browser tab host with the delete button
```diff
diff --git a/src/screens/session.tsx b/src/screens/session.tsx
index 0a7f0a6..c3ba45e 100644
--- a/src/screens/session.tsx
+++ b/src/screens/session.tsx
@@ -1,7 +1,7 @@
 import { Fragment } from 'react'
 import { connect, disconnect } from '~/actions'
 import { Container, LabelledContent } from '~/components'
-import { Box, Button, Separator, Stack, Text } from '~/design-system'
+import { Box, Button, Inline, SFSymbol, Separator, Stack, Text } from '~/design-system'
 import { useHost } from '~/hooks/useHost'
 import { getMessenger } from '~/messengers'
 import { useSessionsStore } from '~/zustand'
@@ -10,8 +10,7 @@ const inpageMessenger = getMessenger('wallet:inpage')
 
 export default function Session() {
   const { data: host } = useHost()
-  const { getSession, getSessions } = useSessionsStore()
-  const sessions = getSessions()
+  const { getSession, sessions } = useSessionsStore()
   const isConnected = Boolean(host && getSession({ host }))
 
   return (
@@ -45,9 +44,32 @@ export default function Session() {
             {Object.values(sessions).map((session) => {
               return (
                 <Fragment key={session.host}>
-                  <LabelledContent label="Host" width="fit">
-                    <Text size="12px">{session.host}</Text>
-                  </LabelledContent>
+                  <Inline
+                    alignVertical="center"
+                    alignHorizontal="justify"
+                    wrap={false}
+                  >
+                    <LabelledContent label="Host" width="fit">
+                      <Text size="12px">{session.host}</Text>
+                    </LabelledContent>
+                    <Box
+                      cursor="pointer"
+                      width="fit"
+                      onClick={() => {
+                        disconnect({
+                          host: session.host,
+                          messenger: inpageMessenger,
+                        })
+                      }}
+                    >
+                      <SFSymbol
+                        color="text/tertiary"
+                        size="12px"
+                        symbol="trash"
+                        weight="medium"
+                      />
+                    </Box>
+                  </Inline>
                   <Separator />
                 </Fragment>
               )
```

Reply:
nice, can use `pbpaste | git apply` but it needed a newline since I got corrupted w.e on the last line?

hmm it seems to work for me, haha? looks good. I made two browser tabs and connected/disconnected on both

https://github.com/paradigmxyz/rivet/assets/136378658/1f2abf07-547a-47b0-945f-f753adbd2bdb

Reply:
Ah, it seems to be related to if you connect to a website that supports a wallet connection or not. I just tried with two dapps locally and that works, but not if I open google.com and hit connect there: google.com still shows up in the list, but disconnecting doesn't trigger a re-render. But that might be ok for now? I'll update this to using `sessions` instead of a getter action and including the delete button, thanks for the help with debugging @ditto-eth! 😄

Reply:
Sorry about that. Seems that the `Record` data structure seemed to play up with our tracked property behavior. 😅

Might be better to represent these as an array. Just pushed up some changes!